### ISSUE TITLE
Linter: parser/semantic ignores `@ts-expect-error`

### ISSUE THREAD

**Issue by hesprs** [oxc-project/oxc#25932]
### What version of Oxlint are you using?

1.79.0

### What command did you run?

`oxlint repro.ts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "options": {
    "typeAware": true
  }
}
```

### What happened?

`repro.ts`:

```TypeScript
export {};

declare global {
  // @ts-expect-error
  module '*.css' {}
}
```

TypeScript `tsc` on `repro.ts` passes without errors, since `TS2435` is suppressed by `@ts-expect-error`.

Oxlint type-aware linting still reports `TS2435`. Even `// oxlint-disable` cannot suppress it.

**Comment by hesprs** [comment]
Just tested, the bug was introduced in `1.78.0` and inherited by `1.79.0`. Pinning Oxlint at 1.77 resolves the bug.

**Comment by camc314** [comment]
Hi, thanks for reporting this issue.

This originates from 0c9b980cb1986b816ec8240ef36d2314ca255cc3, which added the missing TS2435` diagnostic.

This diagnostic is not emitted by tsgo (running with typeaware false should confirm this), but by our own semantic analysis which does not support `@ts-expect-error`, `@ts-ignore`, or Oxlint disable comments.

The right approach here is to fix the invalid TS error by declaring the wildcard module at the top level:
```ts
declare module '*.css' {}
```

Remove the `export {}` and `declare global` wrapper. If you also need global augmentations, place them in a separate `.d.ts` file.

Thanks!

**Comment by hesprs** [comment]
@camc314 If the approaches mentioned by you can fix the issue, I would not use `// @ts-expect-error`. The `repro.ts` is a minimal repro that does not stand for anything.

I'm building a Vite loader that parses `.canvas` file ([source](https://github.com/hesprs/json-canvas-viewer/blob/main/packages/vite/src/index.ts)), package consumers should receive the ambient module augmentation when they import the loader. That's why I need to write the declaration in `.ts` together with runtime code, so the augmentation gets bundled together with the package release.

Nevertheless, a linter (or any type of development tool) in any scenario should provide at least a way to suppress the warning it generates, instead of blocking the CI and leaving no escape hatch.
